app/actions/remote/channel.ts

```diff
@@ -107,6 +108,95 @@ export const fetchChannelByName = async (serverUrl: string, teamId: string, chan
     }
 };
 
+export function generateChannelNameFromDisplayName(displayName: string) {
+    let name = cleanUpUrlable(displayName);
+    if (name === '') {
+        name = generateId();
+    }
+    return name;
+}
+
+export const createChannel = async (serverUrl: string, displayName: string, purpose: string, header: string, type: ChannelType) => {
+    const {database, operator} = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return {error: `${serverUrl} database not found`};
+    }
```

Comment:
if `DatabaseManager.serverDatabases[serverUrl];` is undefined, the deconstruct here will throw an error

```suggestion
    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
    if (!operator) {
        return {error: `${serverUrl} database not found`};
    }
```